### ISSUE TITLE
ctc_decode returns inconsistent result when greedy=False

### ISSUE THREAD

**Issue by angry-andrew** [keras-team/keras#10852]
I am using `ctc_decode` for unsegmented text recognition. When `greedy=True`, `ctc_decode` works as expected. But if `greedy=False`, the function merges repeated classes in the output sequence (for example, ABCCD becomes ABCD).
Depending on the `greedy` flag `ctc_decode` calls either `tf.nn.ctc_greedy_decoder` or `tf.nn.ctc_beam_search_decoder`. Both of them has parameter `merge_repeated`, which is True by default. But the meaning of this parameter is different for these functions.

For `ctc_greedy_decoder`:
```
  If `merge_repeated` is `True`, merge repeated classes in output.
  This means that if consecutive logits' maximum indices are the same,
  only the first of these is emitted.  The sequence `A B B * B * B` (where '*'
  is the blank label) becomes

    * `A B B B` if `merge_repeated=True`.
    * `A B B B B` if `merge_repeated=False`.
```

For `ctc_beam_search_decoder`:
```
  If `merge_repeated` is `True`, merge repeated classes in the output beams.
  This means that if consecutive entries in a beam are the same,
  only the first of these is emitted.  That is, when the top path
  is `A B B B B`, the return value is:

    * `A B` if `merge_repeated = True`.
    * `A B B B B` if `merge_repeated = False`.
```

So, for `ctc_greedy_decoder`, `merge_repeated` works with logits, and for `ctc_beam_search_decoder`, `merge_repeated` works with the result (decoded) sequence.
I tried to set `merge_repeated=False` in `ctc_decode` when `ctc_beam_search_decoder` is called, and `ctc_decode` returned expected result (consistent with the case when `greedy=True`):
```
  if greedy:
    (decoded, log_prob) = ctc.ctc_greedy_decoder(
        inputs=y_pred, sequence_length=input_length)
  else:
    (decoded, log_prob) = ctc.ctc_beam_search_decoder(
        inputs=y_pred,
        sequence_length=input_length,
        beam_width=beam_width,
        top_paths=top_paths,
        merge_repeated=False) 
```

**Comment by bzamecnik** [comment]
Aah, it seems I didn't read your issue thoroughly and made a duplicate: #12238. The fix is easy, I'm gonna make a PR.

**Comment by angry-andrew** [comment]
Thanks! Now I'm sure it wasn't my mistake )
